app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt

```diff
@@ -101,28 +101,29 @@ class BackgroundServices(
             // Notify observers for subscription changes.
             it.registerForSubscriptions(object : PushSubscriptionObserver {
                 override fun onSubscriptionAvailable(subscription: AutoPushSubscription) {
-                    accountManager.authenticatedAccount()?.deviceConstellation()
-                        ?.setDevicePushSubscriptionAsync(
-                            DevicePushSubscription(
-                                endpoint = subscription.endpoint,
-                                publicKey = subscription.publicKey,
-                                authKey = subscription.authKey
+                    // Update for only the services subscription.
+                    if (subscription.type == PushType.Services) {
+                        accountManager.authenticatedAccount()?.deviceConstellation()
+                            ?.setDevicePushSubscriptionAsync(
+                                DevicePushSubscription(
+                                    endpoint = subscription.endpoint,
+                                    publicKey = subscription.publicKey,
+                                    authKey = subscription.authKey
+                                )
                             )
-                        )
+                    }
                 }
             }, ProcessLifecycleOwner.get(), false)
+
+            // We need to force subscribe for nightly users before push server support was added.
+            it.subscribeForType(PushType.Services)
```

Comment:
Looks like you'll run this on every startup. That doesn't sound right?

Reply:
If we don't have a valid subscription, we hit the server, if we return it locally (this is all done in the rust side).